### PR TITLE
Shrink and multiply books

### PR DESCRIPTION
**Pull request by PleasingFungus**
One of the things that makes DCSS exciting is being presented with strategic choices. I started out using long blades, but I found an +8 artefact morningstar at XL 4 - time to switch? I was thinking about going Sif Muna or Vehumet this game, but now I found a Gozag altar on D:2 - what to do?

We'd like for the same to be true of book starts - when to switch in and out of spellcasting, to switch schools, etc. There's currently a large amount of variance that makes this less viable - spellbooks are large, rare, and tend to be internally homogenous, meaning that they either give you a huge pile of relevant spells or none at all.

Instead, shrink spellbooks from an average of about  ~5 spells each to about 3, and generate them about twice as often. Furthermore, change each book to have a wider variety of spell levels and sometimes schools, to increase the odds of them having *something* relevant to any given character. Overall, objstat suggests this keeps total spell generation rates fairly similar.

As part of this, remove the concept of book and spell 'rarity'. Instead, prefer to generate books that have spells appropriate for the current depth. (The numbers here probably need some work.)

Sif Muna and Kiku are mostly untouched. Sif's randbooks stay the same size, and her fixed books get a lower gift timeout.

Spellbook prices in shops are now based on total spell levels. There's probably more work to be done here.

More TODOs:
- Vaults have been minimally adjusted, but theme vaults & alphashops still need to have the new books added, and existing vaults that place custom randbooks need to have their sizes adjusted.
- The new books have no descriptions yet. They either need some, or we need to remove the concept of book descriptions. (That would make me sad, but they're seen so rarely and briefly that I'm not sure I can justify them.)

Feedback welcome!

### CODE REVIEW

**Comment by sdynet** [comment]
Great idea. However, let's leave existing non-random books, like artifact items. I believe a good system is more important than a good lore, but that doesn't mean I don't want any narratives in this game. even though it doesn't matter Orb the Orb of the A, Like we call it the Orb of the Zot.

**Comment by PleasingFungus** [comment]
>  However, let's leave existing non-random books, like artifact items. 

What does this mean? Are you suggesting that existing books (like the Book of Fire, Book of Air etc) should keep their previous contents? If so, how would we accomplish any of the goals described in this PR?

> I believe a good system is more important than a good lore, but that doesn't mean I don't want any narratives in this game.

I don't understand what this sentence has to do with the changes in this PR. Can you explain?

**Comment by sdynet** [comment]
> > However, let's leave existing non-random books, like artifact items.
> 
> What does this mean? Are you suggesting that existing books (like the Book of Fire, Book of Air etc) should keep their previous contents? If so, how would we accomplish any of the goals described in this PR?
> 
> > I believe a good system is more important than a good lore, but that doesn't mean I don't want any narratives in this game.
> 
> I don't understand what this sentence has to do with the changes in this PR. Can you explain?

I understand your writing like this:
"I will delete the existing books and create a random book include two to three spells. The probability of their appearance is twice as likely as it is. In this process, existing books may disappear, or they may lose their own unique explanations."

So after reading this, I was generally in favor of the idea but felt sorry to lose books with attractive explanations. That's why I suggested to you that existing books containing five spells should not touch anything, but make them very rare, just as artifact weapons appear as a very rare possibility.

**Comment by PleasingFungus** [comment]
Ah, no, that's not at all what's happening here, sorry! Existing books are still staying around, they're just losing some of their spells. For example, the Book of Fire will just be Fireball, Starburst, and Ignition. Then, new fixed books with new lore are being added, like the Book of Spheres and the Everburning Encyclopedia.

I hope that helps clear things up!

File: crawl-ref/source/item-name.cc [hunk 1]
```diff
@@ -1108,32 +1125,53 @@ string sub_type_string(const item_def &item, bool known)
...
+            return "Ozocubu's Autobio";
+        case BOOK_YOUNG_POISONERS:
             return "Young Poisoner's Handbook";
-        else if (sub_type == BOOK_FEN)
+        case BOOK_FEN:
             return "Fen Folio";
+        case BOOK_MOON:
+            return "Lunar Lorehold";
+        case BOOK_NEARBY:
+            return "Inescapable Indices";
```
Comment (ebering) [inline crawl-ref/source/item-name.cc:1157]:
```suggestion
            return "Zohar";
```

Less confident about that one. More bike shedding is ongoing.

**Review by ebering — COMMENTED** [review]
The playfeel effects of rarity are something that I'd need many games to get a good handle on and don't currently have time to play sufficiently. Fortunately, per the schedule just discussed in IRC we have a while until freeze, enough time that I feel comfortable trying this out.

I've bikeshed a few book names since I think this patch went overboard with the alliteration.

I also want to suggest one more book

Title: Hunter's Companion
Spells:
Call Canine Familiar
Leda's Liquefaction
Portal Projectile

These three spells are currently at 1 book rarity according to the spreadsheet but imo deserve to be a bit more findable than that. I can add this one post merge.

A bit sillier and less necessary, but for your consideration:
Title: Codex Secundus
Spells: Call Imp
Searing Ray
Corpse Rot

File: crawl-ref/source/item-name.cc [hunk 2]
```diff
@@ -1108,32 +1125,53 @@ string sub_type_string(const item_def &item, bool known)
...
+            return "Everburning Encyclopedia";
+        case BOOK_OZOCUBU:
+            return "Ozocubu's Autobio";
+        case BOOK_YOUNG_POISONERS:
             return "Young Poisoner's Handbook";
-        else if (sub_type == BOOK_FEN)
+        case BOOK_FEN:
             return "Fen Folio";
+        case BOOK_MOON:
+            return "Lunar Lorehold";
```
[the remark this answers is not in the document]
Comment (PleasingFungus) [reply crawl-ref/source/item-name.cc:1155]: Fair re: alliteration, but 'Astrologer' doesn't quite read as a 'book' to me. I'm going with 'Lunar Atlas' for now - open to more ideas/feedback.

File: crawl-ref/source/item-name.cc [hunk 3]
```diff
@@ -1108,32 +1125,53 @@ string sub_type_string(const item_def &item, bool known)
...
+        case BOOK_NEARBY:
+            return "Inescapable Indices";
+        case BOOK_THERE_AND_BACK:
+            return "There-And-Back Book";
+        case BOOK_BIOGRAPHIES_II:
+            return "Great Wizards, Vol. II";
+        case BOOK_BIOGRAPHIES_VII:
+            return "Great Wizards, Vol. VII";
+        case BOOK_TRISMEGISTUS:
+            return "Trismegistus Tome";
```
[the remark this answers is not in the document]
Comment (PleasingFungus) [reply crawl-ref/source/item-name.cc:1165]: Going with 'Trismegistus Codex' here.

File: crawl-ref/source/item-name.cc (hunk 1 above)
Comment (PleasingFungus) [reply crawl-ref/source/item-name.cc:1157]: Will hold off on this for now.

**Pull request closed without merging**